### PR TITLE
WIP: assisted-service: Add zero touch provisioning remote worker day2 jobs

### PR DESCRIPTION
**Pull request by lranjbar**
Add jobs to test remote worker nodes on day 2 to a multi node cluster.

This PR depends on:
https://github.com/openshift/assisted-service/pull/2649
https://github.com/openshift-metal3/dev-scripts/pull/1293

### CODE REVIEW

File: ci-operator/step-registry/baremetalds/assisted/devscripts/remotenodes/baremetalds-assisted-devscripts-remotenodes-commands.sh [hunk 1]
```diff
@@ -0,0 +1,250 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+echo "************ baremetalds assisted devscripts remotenodes command ************"
+## Temp remote_nodes.sh before the dev-scripts PR merges
```
Comment (pawanpinjarkar) [inline ci-operator/step-registry/baremetalds/assisted/devscripts/remotenodes/baremetalds-assisted-devscripts-remotenodes-commands.sh:8]: As the dev-scripts [PR](https://github.com/openshift-metal3/dev-scripts/pull/1293) is merged, this can be removed.

**Comment by carbonin** [comment]
Looks like the day2 jobs are failing with some script error. Is that something that should be resolved here?
```
************ baremetalds assisted devscripts remotenodes command ************
/bin/bash: line 10: CLUSTER_NAME: unbound variable
```

**Comment by lranjbar** [comment]
> Looks like the day2 jobs are failing with some script error. Is that something that should be resolved here?
> 
> ```
> ************ baremetalds assisted devscripts remotenodes command ************
> /bin/bash: line 10: CLUSTER_NAME: unbound variable
> ```

Yup. This is an environment variable that should be set by dev-scripts... it should be present because this is after `baremetalds-devscripts-setup` but sometimes passing these variables between steps is funky. Looks like I'll have to set it again.

**Comment by lranjbar** [comment]
MGMT-7223

File: ci-operator/step-registry/baremetalds/assisted/devscripts/remotenodes/baremetalds-assisted-devscripts-remotenodes-commands.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+timeout -s 9 175m ssh "${SSHOPTS[@]}" "root@${IP}" bash - << EOF |& sed -e 's/.*auths.*/*** PULL_SECRET ***/g'
+
+set -xeuo pipefail
+
+yum install -y git sysstat sos
+systemctl start sysstat
+
+mkdir -p /tmp/artifacts
+
+mkdir dev-scripts
```
Comment (pawanpinjarkar) [inline ci-operator/step-registry/baremetalds/assisted/devscripts/remotenodes/baremetalds-assisted-devscripts-remotenodes-commands.sh:48]:
```
mkdir: cannot create directory ‘dev-scripts’: File exists
{"component":"entrypoint","error":"wrapped process failed: exit status 1","file":"prow/entrypoint/run.go:80","func":"k8s.io/test-infra/prow/entrypoint.Options.Run","level":"error","msg":"Error executing test process","severity":"error","time":"2021-09-22T17:03:26Z"}
error: failed to execute wrapped command: exit status 1
```

**Comment by pawanpinjarkar** [comment]
Looks like there is a syntax error that fails some checks.

File: ci-operator/step-registry/baremetalds/assisted/operator/ztp/day2/baremetalds-assisted-operator-ztp-day2-commands.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  source /root/assisted-ztp-config
+fi
+
+cd "\${REPO_DIR}/deploy/operator/ztp/"
+
+# TODO(lranjbar): Fix the permissions on this file
+#chmod 755 add_day2_remote_nodes.sh
+#./add_day2_remote_nodes.sh
+
+export REMOTE_BAREMETALHOSTS_FILE="${REMOTE_BAREMETALHOSTS_FILE:-/home/test/dev-scripts/ocp/ostest/remote_baremetalhosts.json}"
```
Comment (flaper87) [inline ci-operator/step-registry/baremetalds/assisted/operator/ztp/day2/baremetalds-assisted-operator-ztp-day2-commands.sh:74]: Maybe I'm stating the obvious but, from looking at the logs, it seems like `REMOTE_BAREMETALHOSTS_FILE` is not set when the job gets here and, instead of `REMOTE_BAREMETALHOSTS_FILE` being `/opt/....` it ends up being `/home/test/dev-scripts/ocp/ostest/remote_baremetalhosts.json` which doesn't exist.

Comment (lranjbar) [reply ci-operator/step-registry/baremetalds/assisted/operator/ztp/day2/baremetalds-assisted-operator-ztp-day2-commands.sh:74]: Yes this is correct for this job run. This is the kind of stuff we end up debugging here. The way the values in these variables are set is what is usually the most problematic because it is complex. 

In our jobs we have two environments 1) the pod and 2) the remote server we ssh into. When you set a variable in a job configuration it sets the environment variable in the pod to equal that. However there can be a discrepancy for the environment variable on the remote SSH server. These variables get set every new pod (step) of the job as well.

Here is the official docs: https://docs.ci.openshift.org/docs/architecture/step-registry/#parameters We have added additional complexity with the remote server as well, but the propagation of the variables is complex too.

Comment (lranjbar) [reply ci-operator/step-registry/baremetalds/assisted/operator/ztp/day2/baremetalds-assisted-operator-ztp-day2-commands.sh:74]: This is a thing I've been thinking about a lot. Locally we pipe variables in from previous scripts a lot, the equivalent is steps here. It makes it even more confusing. Essentially in this run it didn't pipe in the right value from `devscripts-remotenodes` which runs before this. Most of the scripts we run locally have this awful piping in from random scripts and places making it hard to follow both locally and it's even worse in the CI.

**Comment by lranjbar** [comment]
The job is passing because I commented out the check for agents. It looks like that the agent is not being created. Will need to debug more on Monday.
